### PR TITLE
Added documentation on the properties for Avro

### PR DESCRIPTION
**Pull request by phillip-kruger**
Initial documentation on the Avro properties for Confluent and Apicurio

Signed-off-by:Phillip Kruger <phillip.kruger@gmail.com>

### CODE REVIEW

File: documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+If you are using https://avro.apache.org/[Apache Avro] serializer/deserializer, please note the following properties:
+
+=== For https://docs.confluent.io/current/schema-registry/serdes-develop/serdes-avro.html[Confluent] Schema Registry
+
+==== Consumer
+
+[options="header"]
+|===
+| "value.deserializer" | "io.confluent.kafka.serializers.KafkaAvroDeserializer"
+| "schema.registry.url" | "http://<your_host>:<your_port>/api/ccompat"
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc:13]: you need CC Compat even on the confluent registry?

File: documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+| "specific.avro.reader" | true
+|===
+
+Example:
+[source, java]
+----
+props.put(org.apache.kafka.clients.consumer.ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, io.confluent.kafka.serializers.KafkaAvroDeserializer.class.getName());
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc:20]: Reactive Messaging does not use this format.

It should be:

```
mp.messaging.incoming.[channel].value.deserializer=io.confluent.kafka.serializers.KafkaAvroDeserializer
mp.messaging.incoming.[channel].schema.registry.url=http://<your_host>:<your_port>/api/ccompat
mp.messaging.incoming.[channel].specific.avro.reader=true
````

File: documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+props.put(io.confluent.kafka.serializers.AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, "http://localhost:8080/api/ccompat"); // Or whatever your host and port is
+props.put(io.confluent.kafka.serializers.KafkaAvroDeserializerConfig.SPECIFIC_AVRO_READER_CONFIG, true);
+----
+
+==== Producer
+
+[options="header"]
+|===
+| "value.serializer" | "io.confluent.kafka.serializers.KafkaAvroSerializer"
+| "schema.registry.url" | "http://<your_host>:<your_port>/api/ccompat"
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc:30]:
```
mp.messaging.outgoing.[channel].value.serializer=io.confluent.kafka.serializers. KafkaAvroSerializer
mp.messaging.outgoing.[channel].schema.registry.url=http://<your_host>:<your_port>/api/ccompat
````

File: documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+|===
+| "value.deserializer" | "io.apicurio.registry.utils.serde.AvroKafkaDeserializer"
+| "apicurio.registry.url" | "http://<your_host>:<your_port>/api"
+| "apicurio.registry.avro-datum-provider" | "io.apicurio.registry.utils.serde.avro.ReflectAvroDatumProvider"
+|===
+
+Example:
+[source, java]
+----
+props.put(org.apache.kafka.clients.consumer.ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, io.apicurio.registry.utils.serde.AvroKafkaDeserializer.class.getName());
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc:54]: Same format as mentioned in my other comment

File: documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+| "apicurio.registry.avro-datum-provider" | "io.apicurio.registry.utils.serde.avro.ReflectAvroDatumProvider"
+| "apicurio.registry.global-id" | "io.apicurio.registry.utils.serde.strategy.GetOrCreateIdStrategy"
+| "apicurio.registry.artifact-id" | "io.apicurio.registry.utils.serde.strategy.SimpleTopicIdStrategy"
+|===
+
+Example:
+[source, java]
+----
+
+props.put(io.apicurio.registry.utils.serde.AbstractKafkaSerializer.REGISTRY_ARTIFACT_ID_STRATEGY_CONFIG_PARAM, io.apicurio.registry.utils.serde.strategy.SimpleTopicIdStrategy.class.getName());
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc:74]: SAme format as mentioned in my other comment

File: documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc (hunk 1 above)
Comment (phillip-kruger) [reply documentation/src/main/doc/modules/kafka/pages/avro-configuration.adoc:13]: O, Yes this is when you use Apicurio but confluent compatible. Let me fix that.

**Comment by cescoffier** [comment]
Thanks!

**Pull request merged**